api/v1alpha1/addon_types.go

```diff
@@ -4,11 +4,25 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 
 	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/variables"
 )
 
+const (
+	AddonStrategyClusterResourceSet AddonStrategy = "ClusterResourceSet"
+	AddonStrategyHelmAddon          AddonStrategy = "HelmAddon"
+
+	VolumeBindingImmediate            = "Immmediate"
+	VolumeBindingWaitForFirstConsumer = "WaitForFirstConsumer"
+
+	VolumeReclaimRecycle = "Recycle"
+	VolumeReclaimDelete  = "Delete"
+	VolumeReclaimRetain  = "Retain"
```

Comment:
Instead of defining our own types lets reuse the upstream ones or set them to those.
```suggestion
	VolumeReclaimRecycle = corev1.PersistentVolumeReclaimRecycle
	VolumeReclaimDelete  = corev1.PersistentVolumeReclaimDelete
	VolumeReclaimRetain  = corev1.PersistentVolumeReclaimRetain
```